invenio_records_resources/services/custom_fields/date.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Custom Fields for InvenioRDM."""
+
+import calendar
+
+from arrow import Arrow
+from edtf import parse_edtf
+from edtf.parser.edtf_exceptions import EDTFParseException
+from marshmallow import fields
+from marshmallow_utils.fields import EDTFDateString, ISODateString
+
+from .base import BaseListCF
+
+
+class ISODateStringCF(BaseListCF):
+    """ISO date custom field."""
+
+    @property
+    def mapping(self):
+        """Return the mapping."""
+        return {"type": "date"}
+
+    @property
+    def field(self):
+        """Marshmallow field custom fields."""
+        _schema = ISODateString(**self._field_args)
+        if self._multiple:
+            return fields.List(_schema)
+        return _schema
+
+
+class EDTFDateStringCF(BaseListCF):
+    """EDTF date custom field."""
+
+    @property
+    def mapping(self):
+        """Return the mapping."""
+        return {
+            "type": "object",
+            "properties": {
+                "date": {"type": "date"},
+                "date_range" : {"type" : "date_range"},
+            }
+        }
+
+    @property
+    def field(self):
+        """Marshmallow field custom fields."""
+        _schema = EDTFDateString(**self._field_args)
+        if self._multiple:
+            return fields.List(_schema)
+        return _schema
+
+    @classmethod
+    def _format_date(cls, date):
```

Comment:
comment: quite some duplication of code with `EDTFDumperExt` (implemented in rdm-records). Could it be moved here and re-used?

Reply:
that would be useful indeed! Is it quick to move?

Reply:
Opened #394